docs/deployment/rds-s3/README.md

```diff
@@ -286,7 +286,81 @@ mysql> select * from observation_logs;
 
 4. Verify the bucket is not empty and was populated by outputs of the experiment.
 
-## 5.0 Uninstall Kubeflow
+## 5.0 (Optional) Setup RDS & S3 credentials access for notebooks & pipelines
+### Use Cases
+- Be able to use `boto3` or aws libraries that require to pass in credentials in a notebook and specify credentials without hard coding them. Access the credentials via environment variables.  
+- Be able to use [ml-metadata](https://github.com/google/ml-metadata/blob/master/g3doc/get_started.md) in a notebook (eg: to explore metadata) and specify the credentials needed by using environment variables.
+- Be able to use [ml-metadata](https://github.com/google/ml-metadata/blob/master/g3doc/get_started.md) to query metadata during a pipeline run by passing a kubernetes secret to a pipeline component.  
+- Be able to use `boto3` or aws libraries that require to pass in credentials in a Kubeflow Pipeline Component. 
+
+The following script takes care of setting up secrets for notebooks and pipelines with values from the secrets manager.  
+Under the hood the script will leverage  [PodDefault resources](https://github.com/kubeflow/kubeflow/blob/master/components/admission-webhook/README.md) to allow to select a configuration with the secrets when creating a notebook. For Kubeflow Pipelines, you would reference the secrets created by the script.  
+The script creates a secret `mysql-secret` secret for the metadata database credentials from RDS and it creates a `mlpipeline-minio-artifact` secret for the AWS S3 credentials. 
+
+### 5.1 Setup the secrets access
+1. Navigate inside the `secrets` directory.
+```
+cd secrets
+```
+2. Install the script dependencies.
+```shell
+pip install -r requirements.txt
+```
+3. Execute the script, replace `YOUR_CLUSTER_REGION`, `YOUR_CLUSTER_NAME` and `YOUR_NAMESPACE` with the appropriate values.  
+Note that `YOUR_NAMESPACE` represents the namespace the secrets will be setup for.  
+For instance if your notebooks and pipelines will be in `kubeflow-user-example-com`, then you would use `kubeflow-user-example-com` in place of `YOUR_NAMESPACE`.
+```shell
+python secrets-setup.py --region YOUR_CLUSTER_REGION --cluster YOUR_CLUSTER_NAME --namespace YOUR_NAMESPACE
+```  
+You can always do `python secrets-setup.py --help` to learn more about the parameters available.    
+
+**Note :** The namespace must exist before executing the script.   
+
+### 5.2 (Optional) Set PodDefault selected by default in notebook UI  
```

Comment:
Is there a generic existing Kubeflow documentation we can point to for this section?